blogposts/from-saas-to-on-prem.md

```diff
@@ -0,0 +1,54 @@
+
+![Weekly Active Users from Customers since 2017](images/on-prem-customerWAUs.jpg "Weekly Active Users from Customers since 2017")
+
+Companies can deploy in-house without exposing their code to Sourcegraph employees or other providers. This allows anyone at a company to start running Sourcegraph locally and hassle-free, instead of needing to ask permission to source out the codebase up to an unauthorized cloud infrastructure provider.
+
+## Ensuring Scalability and Performance
+The performance of locally run software depends mostly on how much hardware the customer is willing to allocate. The benefit for on-premises software here is that, contrary to SaaS, companies aren’t restrained to a one-size-fits-all performance framework, but can scale the software based on the best hardware that best suits their needs and resources. Sourcegraph is able to work with the existing infrastructure of our customers, and scale and perform accordingly. 
+On the flip side, Sourcegraph has to tackle the diverse challenges that come with running a software compatible in different environments.
+
+## All infrastructure is visible
+By running on-premises, our DevOps work is exposed to the customer, making them aware of the quality of our code. From our point of view, this is a blessing in disguise and helps us set our expectations higher for quality and documentation.
+
+Occasionally, things go wrong. One of the challenges of dealing with customer hosted software is that the debugging process requires access to their error logs, alerts, local configuration, deployment environment, and more. To overcome this hurdle, we have improved Sourcegraph’s observability, building [logging and metrics](https://docs.sourcegraph.com/admin/monitoring_and_tracing) into the application to give more information about the root cause of bugs. Fortunately, our customers have been quick to alert us whenever there was an issue.
+
+![Grafana dashboard for monitoring Sourcegraph instance health](images/3.9-grafana-dashboard.png "Grafana dashboard for monitoring Sourcegraph instance health")
+
+## Conclusion
+Switching from SaaS to on-premises was essential for Sourcegraph to become the de facto developer platform that all engineers use among top companies like Lyft, Uber, Plaid, Convoy and many more. Our main challenges are maintaining high standards for our code quality to reduce the slightly more tedious debugging process of an on-premises solution and ensuring Sourcegraph runs smoothly on different environments. The majority of our customers run fairly up to date Sourcegraph versions. We are able to achieve this by making our upgrade process as simple as possible and communicating our enticing new features to our customers. 
+
+![Customer versions](images/on-prem-customerversions.jpg "Customer versions")
```

Comment:
Can we make the title "Current customer versions (as of October 2019)" or similar to clarify that this is a snapshot when the blog post was published?

Reply:
Yes.